File: src/pset2/caesar.c
#include<stdio.h>
#include<cs50.h>
#include<ctype.h>
#include<string.h>
#include<math.h>

int main(int argc, string argv[]){
    if(argc != 2){
        printf("Try again and type in a number in the command prompt\n");
        return 1;
    }
    string x = GetString();
    int k = atoi(argv[1]);
    for(int i = 0; i < strlen(x); i++){
        if(isalpha(x[i])){
        k = k % 26;
        unsigned char v = x[i] + k;
        int y = k / 26;
        char u = v - k;
        if(u < 91 && v > 90){
            v -= ((floor(y) * 26) + 26);
        }
        if((u > 96 && u < 123) && v > 122){
            v -= ((floor(y) * 26) + 26);
        }
        printf("%c", v);
        }
        else{
            printf("%c", x[i]);
        }
    }
    printf("\n");
    
}

File: src/pset6/caesar.py
import sys
from math import *
if len(sys.argv) != 2:
    print("Try again and type in a number in the command prompt")
    exit(1)

x = input("")
k = int(sys.argv[1])
for i in range(len(x)):
    if x[i].isalpha():
        k = k % 26
        v = chr(ord(x[i]) + k)
        y = k / 26
        u = chr(ord(v) - k)
        if ord(u) < 91 and ord(v) > 90:
            v = chr(ord(v) - ((floor(y) * 26) + 26))
        if (ord(u) > 96 and ord(u) < 123) and ord(v) > 122:
            v = chr(ord(v) - ((floor(y) * 26) + 26))
        print(v, end = "")
    else:
        print(x[i], end = "")
print("")
